src/Models/Sources.php

```diff
@@ -102,4 +104,52 @@ public function validateAppleUser(string $identityToken): object
 
         return $appleUserInfo;
     }
+
+    /**
+     * validation.
+     *
+     * @param  string $email
+     * @param  string $token
+     *
+     * @return bool
+     */
+    public function validation(string $email, string $token) : bool
+    {
```

Comment:
@FredPeal let's use constants for google and fb

Let's have the function continue throwing an exception instead of just returning false

how do we test?